nixos/modules/services/backup/restic.nix

```diff
@@ -121,6 +149,29 @@ with lib;
           extraOptions = concatMapStrings (arg: " -o ${arg}") backup.extraOptions;
           connectTo = elemAt (splitString ":" backup.repository) 1;
           resticCmd = "${pkgs.restic}/bin/restic${extraOptions}";
+          filesFromTmpFile = "/tmp/restic-backup-${name}-files";
```

Comment:
Redirecting to files in `/tmp` is dangerous this way. The target could be replaced by unprivileged users with a symlink and overwrite arbitrary files on the system. Either use systemd's RuntimeDirectory (preferred option) or use `mktemp` followed by a proper cleanup.

Reply:
hm ... `PrivateTmp` sounds better than `RuntimeDirectory`, no?  (Didn't know about either of them until you mentioned it, though -- thanks!)

Reply:
They are not mutual exclusive, but RuntimeDirectory has the advantage that root can at least without tricks access those. PrivateTmp is more meant for applications that we cannot not teach to use a different directory, while for RuntimeDirectory, each daemon gets its own path to use.